test/OpenTelemetry.Exporter.OneCollector.Tests/WriteDirectlyToTransportSinkTests.cs

```diff
@@ -0,0 +1,245 @@
+        Assert.Equal("\"item3\"\n\"item4\"\n", Encoding.ASCII.GetString(data));
+
+        Assert.NotEqual(0, sink.Buffer.Length);
+
+        sink.Dispose();
+
+        Assert.Equal(0, sink.Buffer.Length);
+
+        data = transport.ExportedData[2];
+
+        Assert.NotNull(data);
+        Assert.Equal("\"item5\"\n", Encoding.ASCII.GetString(data));
+    }
+
+    private sealed class TestSerializer : CommonSchemaJsonSerializer<string>
+    {
+        public TestSerializer(
+            int maxPayloadSizeInBytes = int.MaxValue,
+            int maxNumberOfItemsPerPayload = int.MaxValue)
+            : base("tenant-token", maxPayloadSizeInBytes, maxNumberOfItemsPerPayload)
+        {
+        }
+
+        public override string Description => nameof(TestSerializer);
+
+        protected override void SerializeItemToJson(Resource resource, string item, Utf8JsonWriter writer)
+        {
+            writer.WriteStringValue(item);
+        }
+    }
+
+    /*
+    private sealed class TestSerializer : ISerializer<string>
+    {
+        private readonly int maxPayloadSizeInBytes;
+        private readonly int maxNumberOfItems;
+
+        public TestSerializer(int maxNumberOfItems = int.MaxValue)
+        {
+            this.maxNumberOfItems = maxNumberOfItems;
+        }
+
+        public string Description => nameof(TestSerializer);
+
+        public void SerializeBatchOfItemsToStream(
+            Resource resource,
+            in Batch<string> batch,
+            Stream stream,
+            out BatchSerializationResult serializationResult)
+        {
+            var numberOfItemsSerialized = 0;
+
+            stream.WriteByte((byte)'[');
+
+            foreach (var item in batch)
+            {
+                if (numberOfItemsSerialized > 0)
+                {
+                    stream.WriteByte((byte)',');
+                }
+
+                stream.WriteByte((byte)'"');
+
+                byte[] data = Encoding.ASCII.GetBytes(item);
+                stream.Write(data, 0, data.Length);
+
+                stream.WriteByte((byte)'"');
+
+                if (numberOfItemsSerialized++ >= this.maxNumberOfItems)
+                {
+                    break;
+                }
+
+                if (payloadSizeInBytes >= this.maxPayloadSizeInBytes)
+                {
+                    result = new BatchSerializationResult
+                    {
+                        NumberOfItemsSerialized = numberOfSerializedItems,
+                        PayloadSizeInBytes = payloadSizeInBytes,
+                        PayloadOverflowItemSizeInBytes = currentItemSizeInBytes,
+                    };
+                    return;
+                }
+            }
+
+            stream.WriteByte((byte)']');
+
+            serializationResult = new BatchSerializationResult
+            {
+                NumberOfItemsSerialized = numberOfItemsSerialized,
+                PayloadSizeInBytes = stream.Length,
+            };
+        }
+    }
+    */
+
```

Comment:
Do you rellay need this code?

```suggestion
```

Reply:
Woops sorry it was used at one point but I refactored it out. Removed 😄